File: solutions/fsharp/log-levels/1/LogLevels.fs
module LogLevels

let message (logLine: string): string = 
    let startIndex = logLine.IndexOf ":" + 2
    logLine.[startIndex..].Trim()

let logLevel(logLine: string): string = 
    let endIndex = logLine.IndexOf ":" - 2
    logLine.[1..endIndex].ToLower()

let reformat(logLine: string): string = 
    let msg = message logLine
    let level = logLevel logLine
    sprintf "%s (%s)" msg level

File: solutions/csharp/log-levels/1/LogLevels.cs
static class LogLine
{
    public static string Message(string logLine)
    {
        string message = logLine.Split("]: ")[1].Trim();
        return message;
    }

    public static string LogLevel(string logLine)
    {
        int start = logLine.IndexOf("[");
        int end = logLine.IndexOf("]");
        return logLine.Substring(start + 1, end - 1).ToLower();
    }

    public static string Reformat(string logLine)
    {
        return $"{LogLine.Message(logLine)} ({LogLine.LogLevel(logLine)})";
    }
}
